yext/answers-search-ui
# build_and_deploy_i18n workflow

add a new workflow for build and deploy internationalization assets. 
- for develop branch: only run canary deployments (canary/latests and canary/[github.sha])
- for non-develop branch within the restricted set of branches specified in trigger event: run normal deployment (dev/[branch-name])
- Also pull the process for formatting branch name into its own reusable workflow to avoid duplicating that code portion for build and deploy workflows.

J=SLAP-1817
TEST=manual

- check in s3 bucket for this branch and see that i18n assets are deployed.
- check github actions for normal build_and_deploy and build_and_deploy_i18n, see that both works with the new format_branch_name refactor
- with canary build and conditional deployment base on specific branch name, see that only canary deployment jobs ran. See in s3 buckets, the two set of canary assets in the corresponding mock directories base on this branch are present. https://github.com/yext/answers-search-ui/actions/runs/1762091174
